src/dependent_read.c

```diff
@@ -0,0 +1,30 @@
+// Copyright (c) Microsoft Corporation
+// SPDX-License-Identifier: MIT
+
+#include "bpf.h"
+
+__attribute__((section("xdp"), used)) void
+dependent_read(struct xdp_md* ctx)
+{
+    // Set r0 to 0 avoid returning uninitialized register.
+    asm volatile("r0 = 0");
+
+    // Load data into r7
+    asm volatile("r7 = %0" : : "r"(ctx->data));
+    // Load data_end into r8
+    asm volatile("r8 = %0" : : "r"(ctx->data_end));
+    // Compute packet length and store in r8.
```

Comment:
Good that this comment ends in punctuation but other lines don't.
Would be good to be consistent.